crates/holochain_terminal/src/components/bootstrap.rs

```diff
@@ -0,0 +1,229 @@
+            ScreenEvent::SwitchNetwork => {
+                switch_network = true;
+                refresh = true; // Always refresh when switching network
+                *LAST_REFRESH_AT.write().unwrap() = Some(Instant::now()); // Reset the refresh timer
+            }
+            ScreenEvent::NavDown => {
+                let mut selected = SELECTED.write().unwrap();
+                let agents = AGENTS.read().unwrap();
+
+                if *selected < agents.len() - 1 {
+                    *selected += 1;
+                }
+            }
+            ScreenEvent::NavUp => {
+                let mut selected = SELECTED.write().unwrap();
+
+                if *selected > 0 {
+                    *selected -= 1;
+                }
+            }
+        }
+    }
+
+    if switch_network {
+        let mut network_type = NETWORK_TYPE
+            .write()
+            .expect("Should have been able to read network type");
+
+        let new_net = match *network_type {
+            BootstrapNet::Tx2 => BootstrapNet::Tx5,
+            BootstrapNet::Tx5 => BootstrapNet::Tx2,
+        };
+
+        *network_type = new_net;
+    }
+
+    if refresh {
+        *SELECTED.write().unwrap() = 0;
+
+        match block_on(
+            async {
+                let network_type = { *NETWORK_TYPE.read().unwrap() };
+                random(
+                    Some(bootstrap_url.into()),
+                    RandomQuery {
+                        // TODO This conversion is defined but it's in holochain_p2p which shouldn't be a dep of this crate.
+                        space: Arc::new(KitsuneSpace::new(dna_hash.get_raw_36().to_vec())),
+                        limit: RandomLimit(30),
+                    },
+                    network_type,
+                )
+                .await
+            },
+            std::time::Duration::from_secs(10),
+        ) {
+            Ok(Ok(agents)) => {
+                *AGENTS.write().unwrap() = agents;
+            }
+            Ok(Err(e)) => {
+                show_message(
+                    format!("Error fetching agents - {:?}", e).as_str(),
+                    frame,
+                    rect,
+                );
+                return;
+            }
+            Err(_) => {
+                show_message("Timeout while fetching agents", frame, rect);
+                return;
+            }
+        };
+    }
+
+    let agents = AGENTS.read().expect("Should have been able to read agents");
+
+    let screen_layout = Layout::default()
+        .direction(Direction::Vertical)
+        .constraints([Constraint::Min(0), Constraint::Length(1)])
+        .split(rect);
+
+    let content_layout = Layout::default()
+        .direction(Direction::Horizontal)
+        .constraints([Constraint::Percentage(50), Constraint::Percentage(50)])
+        .split(screen_layout[0]);
+
+    let list_items: Vec<ListItem> = agents
+        .iter()
+        .map(|a| ListItem::new(format!("Agent {:?}", a.agent)))
```

Comment:
Do you want to use the base64 version here too, or do you need the hex for some reason?

Reply:
I do want this to match! Done :)